skaha/src/main/java/org/opencadc/skaha/session/GetAction.java

```diff
@@ -95,19 +101,66 @@ public GetAction() {
     public void doAction() throws Exception {
         super.initRequest();
         if (requestType.equals(REQUEST_TYPE_SESSION)) {
+            String view = syncInput.getParameter("view");
             if (sessionID == null) {
-                // List the sessions
-                String typeFilter = syncInput.getParameter("type");
-                String statusFilter = syncInput.getParameter("status");
-                String view = syncInput.getParameter("view");
-                boolean allUsers = SESSION_LIST_VIEW_ALL.equals(view);
+                if (SESSION_VIEW_STATS.equals(view)) {
+                    // report stats on sessions and resources
+                    List<Session> sessions = getAllSessions(null);
+                    int desktopCount = filter(sessions, "desktop-app", "Running").size();
+                    int headlessCount = filter(sessions, "headless", "Running").size();
+                    int totalCount = sessions.size();
+                    String k8sNamespace = getNamespace();
+                    try {
+                        int coresInUse = 0;
+                        int coresAvailable = 0;
+                        int maxCores = 0;
+                        int maxRAM = 0;
+                        int withCores = 0;
+                        int withRAM = 0;
+                        List<String> nodeNames = getNodeNames(k8sNamespace);
+                        for (String nodeName : nodeNames) {
+                            int rCPUCores = getCPUCores(nodeName, k8sNamespace);
+                            int aResources[] = getAvailableResources(nodeName, k8sNamespace);
+                            int aCPUCores = aResources[0];
+                            if (aCPUCores > maxCores) {
+                                maxCores = aCPUCores;
+                                withRAM = aResources[1];
+                            }
+                            
+                            int aRAM = aResources[1];
+                            if (aRAM > maxRAM) {
+                                maxRAM = aRAM;
+                                withCores= aCPUCores;
+                            }
+                            
+                            coresInUse = coresInUse + rCPUCores;
+                            coresAvailable = coresAvailable + aCPUCores;
+                            log.debug("Node: " + nodeName + " Cores: " + rCPUCores + "/" + aCPUCores + " RAM: " + aRAM + " GB");
+                        }
+
```

Comment:
This is looking better, but you could move the bulk of the stats collection to it's own method?  doAction() is too busy now.

Reply:
Refactored doAction.